app/Scrambler.py

```diff
@@ -0,0 +1,64 @@
+# pylint: disable-all
+
+from random import choice, shuffle
+
+from Picture import Picture
+from Tile import Tile
+
+
+class Scrambler:
+    @classmethod
+    def scramble_rows(cls, picture: Picture) -> None:
+        """Splits an Image up into rows an rearranges them."""
+        num_tiles = 8
+        image_size = picture.image.size
+        tile_size = (image_size[0], int(image_size[1] / num_tiles))
+
+        for y in range(num_tiles):
+            x0, y0, x1, y1 = (0, tile_size[1] * y, tile_size[0], tile_size[1] * (y + 1))
+            picture.tiles[y] = Tile(picture.image.crop((x0, y0, x1, y1)), (x0, y0))
+
+        picture.tile_order = list(picture.tiles.keys())
+        shuffle(picture.tile_order)
+
+    @classmethod
+    def scramble_grid(cls, picture: Picture) -> None:
+        """Splits an Image up into tiles an rearranges them and rotates each tile randomly."""
+        num_of_tiles = 4
+        co_ordinates = [(0, 0, 512, 512), (512, 0, 1024, 512), (0, 512, 512, 1024), (512, 512, 1024, 1024)]
+
+        # Tile swap logic
+        tile_order = [0, 1, 2, 3]
+        swap_list = ["top", "left", "bottom", "right"]
+        tile_rotations = [choice([90, 180, 270]) for _ in range(num_of_tiles)]
+        swap_order = []
+
+        shuffle(swap_list)
+        for current_choice in swap_list:
+            swap_order.append(current_choice)
+            if current_choice == "top":
+                tile_order[0], tile_order[1] = tile_order[1], tile_order[0]
+
+            if current_choice == "left":
+                tile_order[0], tile_order[2] = tile_order[2], tile_order[0]
+
+            if current_choice == "bottom":
+                tile_order[2], tile_order[3] = tile_order[3], tile_order[2]
+
+            if current_choice == "right":
+                tile_order[1], tile_order[3] = tile_order[3], tile_order[1]
+
+        # Order of tiles to be formed
+        picture.tile_order = tile_order
+
+        # Adding tiles for picture
+        for tile in range(num_of_tiles):
+            picture.tiles[tile] = Tile(
```

Comment:
I see you pick rotations for each tile, but I don't understand how the tiles you create here are actually rotated. Could you please put in a comment explaining how that works?

Reply:
Tiles were being rotated in the save function of Picture.py as discussed earlier to keep the save function same for all scramblers rotation is being done in Scrambler.py itself.